misc_docs/syntax/decorator_uncurry.mdx

```diff
@@ -0,0 +1,34 @@
+---
+id: "uncurry-decorator"
+keywords: ["uncurry", "decorator"]
+name: "@uncurry"
+summary: "This is the `@uncurry` decorator."
+category: "decorators"
+---
+
+The `@uncurry` decorator allows you to annotate **external** function callback argument types
+so when writing the callbacks they may be declared as normal functions `() => { ... }`
+rather than needing the uncurried function syntax `(.) => { ... }`.
+
+### Example
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res
+@bs.send
+external map: (array<'a>, @uncurry ('a => 'b)) => array<'b> = "map"
+
+let result = map([1, 2, 3], x => x + 1)
+```
+
+```js
+var result = [1, 2, 3].map(function (x) {
+  return (x + 1) | 0
+})
+```
+
```

Comment:
As we can see, we defined a regular function `('a) => 'b` instead of `(. 'a) => 'b`, but still have all the guarantees. Please note that the compiler does a lot of optimizations, so the example above will compile to the same code even without the `@uncurry` decorator. Adding the decorator (or using the `(.) =>` syntax) makes the output 100% predictable though.

Reply:
That makes it clearer, thanks.